### PR TITLE
Throw on config change post-activation

### PR DESCRIPTION
**Pull request by chrisdunelm**
Fixes 1036

### CODE REVIEW

File: apis/Google.Cloud.Logging.Log4Net/Google.Cloud.Logging.Log4Net.Tests/Log4NetTest.cs [hunk 1]
```diff
@@ -179,6 +179,52 @@ private LoggingEvent CreateLog(Level level, string msg, int secondsOfs = 0)
             .ToString(GoogleStackdriverAppender.s_lostDateTimeFmt);
 
         [Fact]
+        public async Task PostActivationChangeThrow()
+        {
+            await RunTestWorkingServer(appender =>
+            {
+                Assert.Throws<InvalidOperationException>(() => { appender.DisableResourceTypeDetection = true; });
```
Comment (jskeet) [inline apis/Google.Cloud.Logging.Log4Net/Google.Cloud.Logging.Log4Net.Tests/Log4NetTest.cs]: I don't think you should need the braces here... assignments can still be expression lambdas.

File: apis/Google.Cloud.Logging.Log4Net/Google.Cloud.Logging.Log4Net/GoogleStackdriverAppender_Configuration.cs [hunk 2]
```diff
@@ -177,62 +201,96 @@ public void AddResourceLabel(Label label)
...
-        public int DisposeTimeoutSeconds { get; set; } = 30;
+        public int DisposeTimeoutSeconds
+        {
+            get => _disposeTimeoutSeconds;
+            set => _disposeTimeoutSeconds = ThrowIfActivated(value, nameof(DisposeTimeoutSeconds));
+        }
+
+        private T ThrowIfActivated<T>(T value, string name)
+        {
+            GaxPreconditions.CheckState(!_isActivated, $"Appender already activated; cannot modify '{name}'");
```
Comment (jskeet) [inline apis/Google.Cloud.Logging.Log4Net/Google.Cloud.Logging.Log4Net/GoogleStackdriverAppender_Configuration.cs]: Tiny nit:

```csharp
GaxPreconditions.CheckState(!_isActivated, "Appender already activated; cannot modify '{0}'", name);
```

... that avoids the string formatting for non-errors. Admittedly this is unlikely to be a performance bottleneck, but...

File: apis/Google.Cloud.Logging.Log4Net/Google.Cloud.Logging.Log4Net.Tests/Log4NetTest.cs (hunk 1 above)
Comment (chrisdunelm) [inline apis/Google.Cloud.Logging.Log4Net/Google.Cloud.Logging.Log4Net.Tests/Log4NetTest.cs]: It appears you're right! You *do* know C#! Who'd have thought it??

File: apis/Google.Cloud.Logging.Log4Net/Google.Cloud.Logging.Log4Net/GoogleStackdriverAppender_Configuration.cs (hunk 2 above)
Comment (chrisdunelm) [inline apis/Google.Cloud.Logging.Log4Net/Google.Cloud.Logging.Log4Net/GoogleStackdriverAppender_Configuration.cs]: I'd decided I didn't care; but changed.

**Pull request merged**